src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Uap.cs

```diff
@@ -15,4 +18,50 @@ internal static partial class ProcessManager
     {
         public static IntPtr GetMainWindowHandle(int processId) => IntPtr.Zero;
     }
+
+    internal static partial class NtProcessManager
+    {
+        private static ProcessModuleCollection GetModules(int processId, bool firstModuleOnly)
+        {
+            // We don't have a good way of getting all of the modules of the particular process,
+            // but we can at least get the path to the executable file for the process, and
+            // other than for debugging tools, that's the main reason consumers of Modules care about it,
+            // and why MainModule exists.
+            try
+            {
+                ArrayPool<char> pool = ArrayPool<char>.Create(maxArraysPerBucket: 1, maxArrayLength: 64);
```

Comment:
If I understand right, Create makes your own pool. So I don't think it adds any value over just newing up your own char array and reusing it for each call to GetModuleFileNameEx. To get value from pooling in this context, you'll want to do `ArrayPool<char>.Shared.Rent(64);` ie

```
chars = ArrayPool<char>.Shared.Rent(64);
.. use chars for several pinvokes then finally ...
ArrayPool<char>.Shared.Return(chars);
````
That way the hope is you're using an array someone else newed up, or they can reuse yours.